File: Lib/Telerik/Examples/QuickStart/Scheduler/CustomAppointmentDialog/CustomAppointmentFactory.vb
Imports Telerik.WinControls.UI

Namespace Telerik.Examples.WinControls.Scheduler.CustomAppointmentDialog
	Public Class CustomAppointmentFactory
		Implements IAppointmentFactory
#Region "IAppointmentFactory Members"

        Public Function CreateNewAppointment() As IEvent Implements IAppointmentFactory.CreateNewAppointment
            Return New OutlookLikeAppointment()
        End Function

#End Region
    End Class
End Namespace

File: Lib/Telerik/Examples/QuickStart/Scheduler/CustomAppointmentDialog/CustomAppointmentFactory.cs
using Telerik.WinControls.UI;

namespace Telerik.Examples.WinControls.Scheduler.CustomAppointmentDialog
{
    public class CustomAppointmentFactory : IAppointmentFactory
    {
        #region IAppointmentFactory Members

        public IEvent CreateNewAppointment()
        {
            return new OutlookLikeAppointment();
        }

        #endregion
    }
}
